core/common/src/main/java/alluxio/grpc/GrpcManagedChannelPool.java

```diff
@@ -261,31 +170,78 @@ private ManagedChannel createManagedChannel(GrpcChannelKey channelKey) {
     return channelBuilder.build();
   }
 
+  /**
+   * Returns {@code true} if given managed channel is ready.
+   */
+  private boolean waitForChannelReady(ManagedChannel managedChannel, long healthCheckTimeoutMs) {
+    try {
+      Boolean res = CommonUtils.waitForResult("channel to be ready", () -> {
+        ConnectivityState currentState = managedChannel.getState(true);
+        switch (currentState) {
+          case READY:
+            return true;
+          case TRANSIENT_FAILURE:
+          case SHUTDOWN:
+            return false;
+          case IDLE:
+          case CONNECTING:
+            return null;
+          default:
+            return null;
+        }
+      }, WaitForOptions.defaults().setTimeoutMs((int) healthCheckTimeoutMs));
```

Comment:
Should we set the interval based on the timeout?

Reply:
I think the default (20ms) is what we want as we'd like to get out out of this ASAP.